### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by depombo**
https://docs.anycloudapp.com/v/v0.1.5/

### CODE REVIEW

File: docs/about.md
Comment (dfellis) [inline docs/about.md:12]: Should remove the extra blank line here.

File: docs/host-node-aws.md
Comment (dfellis) [inline docs/host-node-aws.md:9]: I assume this is to be continued

File: docs/how-to-guides/env.md
Comment (dfellis) [inline docs/how-to-guides/env.md:3]: Exactly how I feel about environment variables!

File: docs/how-to-guides/deploy-to-azure.md
Comment (dfellis) [inline docs/how-to-guides/deploy-to-azure.md:2]: Since there's nothing here, I guess the recommendation is: "don't"? ;)

File: docs/reference-1/credentials-1.md
Comment (dfellis) [inline docs/reference-1/credentials-1.md:61]: Why the name `stripe-*` on these? It's a bit confusing and makes me think you might need a Stripe account, too?

File: docs/reference-1/project-structure/README.md
Comment (dfellis) [inline docs/reference-1/project-structure/README.md:7]: Not `git add -A`?

File: docs/reference/credentials.md
Comment (dfellis) [inline docs/reference/credentials.md:1]: Why are these files defined twice?

File: docs/how-to-guides/azure.md
Comment (cdmistman) [inline docs/how-to-guides/azure.md:5]: nit:
```suggestion
1. Create an Azure Active Directory application and its Service Principal as described [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal). After following the previous instructions you can copy your `Application (client) ID` and your `Directory (tenant) ID`.
```

File: docs/reference-1/credentials-1.md
Comment (depombo) [reply docs/reference-1/credentials-1.md:61]: Oh, that was just the name of an example company. I will use `piedpiper-*` 😂

File: docs/host-node-aws.md
Comment (depombo) [reply docs/host-node-aws.md:9]: Yep!

File: docs/how-to-guides/deploy-to-azure.md
Comment (depombo) [reply docs/how-to-guides/deploy-to-azure.md:2]: hahaha

File: docs/reference/credentials.md
Comment (depombo) [reply docs/reference/credentials.md:1]: Good catch, fixed!

File: docs/reference/structure/anycloud-json.md
Comment (dfellis) [inline docs/reference/structure/anycloud-json.md:38]: Accidental reversion here.

File: docs/reference/structure/anycloud-json.md
Comment (dfellis) [inline docs/reference/structure/anycloud-json.md:44]: And here

File: docs/about.md
Comment (dfellis) [inline docs/about.md:40]: We should look into [Chocolatey](https://chocolatey.org/) for Windows here.

File: docs/about.md
Comment (dfellis) [inline docs/about.md:33]: Making a `.deb` and a `.rpm` sounds like a decent idea here, perhaps setting up a [PPA](https://help.ubuntu.com/community/PPA)

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:41]: Technically it's using the port it received from an environment variable.

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:22]: This bit *really* makes me wish we had an `anycloud config` sub-command with perhaps interactive prompting of what provider, what credentials, and what alias to provide.

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:99]: I don't think you want to define the name "staging" as the deploy name here. Anyone copy-pasting these commands will likely collide with *someone else* who has copy-pasted the command and the "staging" name is taken.

It's probably better to just leave this as `anycloud new` and that's it.

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:93]: The missing 7th step is that all of the changes in the git repo *must* be committed or they won't be deployed.

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:114]: Two things:

1. The actual version they'll see is v0.1.35 as of right now
2. This column actually doesn't make any sense to the anycloud users. Should we even have it?

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:134]: "Terminate your AnyCloud app when you no longer need it"? The current phrasing somehow makes it seem like they're supposed to be temporary?

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:99]: Can we add some explanation that the "staging" here must match a key in the `anycloud.json` file?

File: docs/how-to/custom-app.md
Comment (dfellis) [inline docs/how-to/custom-app.md:6]: People may come into this page directly, not having read the Node express hello world tutorial, so it would probably be good to explain what `staging` is referring to.

File: docs/how-to/env.md
Comment (dfellis) [inline docs/how-to/env.md:3]: It would be good to add an extra sentence that the file consists of `KEY=VALUE` pairs on separate lines.

Comment (dfellis) [reply docs/how-to/env.md:3]: Or to link to, say, Bash or Docker documentation on how they should be formatted.

File: docs/how-to/multi.md
Comment (dfellis) [inline docs/how-to/multi.md:1]: I think it would really be worth it to have a single paragraph right below this title that sells up the ability of Anycloud to deploy a singular logical cluster to multiple regions and/or multiple clouds at the same time, and that it will always try to keep at least one running server in each region/cloud defined.

File: docs/reference/credentials.md
Comment (dfellis) [inline docs/reference/credentials.md:35]: I think you can condense this quite a bit and get everything across:

```json
{
  "piedpiper-aws": {
    "cloudProvider": "AWS",
    "credentials": {
      "accessKeyId": "#############",
      "secretAccessKey": "#############"
    }
}
```

The top-level key is the alias you provide for referring to these credentials. For AWS the `cloudProvider` value is `AWS`, and in the credentials the `accessKeyId` and `secretAccessKey` come from an IAM user with an [`AmazonEC2FullAccess`](https://console.aws.amazon.com/iam/home#/policies/arn%3Aaws%3Aiam%3A%3Aaws%3Apolicy%2FAmazonEC2FullAccess)policy attached.


This takes up less space while explaining more at the same time, and the other examples below can also be reworked similarly.

File: docs/reference/credentials.md
Comment (dfellis) [inline docs/reference/credentials.md:32]: Also, you can't have trailing commas in JSON files, so we shouldn't have those in our examples. ;)

File: docs/background/autoscaling.md
Comment (dfellis) [inline docs/background/autoscaling.md:3]: So my interpretation of [the Divio documentation system](https://documentation.divio.com/introduction/) is that this shouldn't be "background information." It should be "Explanations" on things like "Why should I use Anycloud?" "Why *shouldn't* I use Anycloud?" "What complications do multi-region and/or multi-cloud deployments have?" etc.

I think short essay-level articles on things like that are more useful for end users and closer to the intent of this "quadrant."

File: docs/about.md
Comment (cdmistman) [inline docs/about.md:35]: We should explicitly state PowerShell so users don't try this in cmd.

File: docs/how-to/azure.md
Comment (cdmistman) [reply docs/how-to/azure.md:13]: I think Pied Piper, considering it's [not a real company](https://youtu.be/oTD-x9tJa2M)

File: docs/about.md
Comment (depombo) [reply docs/about.md:33]: Yes, added that to https://github.com/alantech/meta/issues/147 which is the issue that tracks that. All users so far are MacOS so it has not been prioritized. Also this is not ideal, but it would work

File: docs/about.md
Comment (depombo) [reply docs/about.md:40]: Yes, added that to https://github.com/alantech/meta/issues/147 which is the issue that tracks that

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:93]: ah yes!

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:114]: This table is the same for `alan deploy` as for `anycloud`. We could remove it in the former case, but then we add more special casing. Also this *might* make debugging for us easier when a user is having a problem and it could also prompt the users to upgrade.

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:22]: Yeah, this has been on my mind. Instead of another subcommand. In my head, it would go into the interactive prompting whenever the configuration is not properly set up.

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:99]: Yes!

File: docs/how-to/custom-app.md
Comment (depombo) [reply docs/how-to/custom-app.md:6]: yes, thanks!

File: docs/how-to/env.md
Comment (depombo) [reply docs/how-to/env.md:3]: yes

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:14]: hmm I will use David's suggestion in `credentials.md` to say "The top-level key is the alias you provide for referring to these credentials in `anycloud.json`."

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:26]: Ahh good catch!

File: docs/how-to/gcp.md
Comment (depombo) [reply docs/how-to/gcp.md:13]: yes, will change to `mystartup-*`

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md]: Sounds good, I will set it in the code to keep it simple!

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:3]: Ah yes, those will only work once this is landed and `v0.1.5` becomes the "main" docs branch

File: docs/reference/project-structure.md
Comment (depombo) [reply docs/reference/project-structure.md:13]: I think it is fine now, because they are all in the same file just different section so the user only has to scroll

File: docs/reference/structure/anycloud-json.md
Comment (depombo) [reply docs/reference/structure/anycloud-json.md]: Yeah gitbook was acting up / I wasn't using it properly

File: docs/reference/project-structure.md
Comment (cdmistman) [reply docs/reference/project-structure.md:13]: maybe link to the section then? Feels weird having a link that doesn't go anywhere...

File: docs/tutorial.md
Comment (dfellis) [inline docs/tutorial.md:22]: Still have a trailing comma that will fail JSON parsing. Also, can you change the ` ```javascript` to ` ```json` and see if that works? It does work on Github and it properly highlights invalid JSON for you:

```json
{
  "some: "JSON"
}
```

Comment (depombo) [reply docs/tutorial.md:22]: yeah and `json` doesn't work on Gitbook

File: docs/tutorial.md
Comment (dfellis) [reply docs/tutorial.md:102]: Now, should this note the first-time Github login that's being done?

File: docs/tutorial.md
Comment (dfellis) [reply docs/tutorial.md:106]: Still broken

File: docs/summary.md
Comment (dfellis) [reply docs/summary.md:10]: +1 on this, still missing the `Deploy to AWS` section. It's mostly copy-paste from the tutorial, but it should be explicitly called out separately for someone looking for it that doesn't want to go through "the whole tutorial"

File: docs/reference/project-structure.md
Comment (dfellis) [inline docs/reference/project-structure.md:48]: Nit: trailing comma

File: docs/reference/project-structure.md
Comment (dfellis) [inline docs/reference/project-structure.md:54]: Nit: trailing comma

File: docs/reference/project-structure.md
Comment (dfellis) [reply docs/reference/project-structure.md:48]:
```suggestion
    "credentials": "piedpiper-aws"
```

File: docs/reference/project-structure.md
Comment (dfellis) [reply docs/reference/project-structure.md:54]:
```suggestion
    "credentials": "piedpiper-aws"
```

File: docs/tutorial.md
Comment (dfellis) [reply docs/tutorial.md:22]:
```suggestion
    }
```

File: docs/how-to/multi.md
Comment (dfellis) [inline docs/how-to/multi.md:37]:
```suggestion
      "vmType": "t3.medium"
```

File: docs/how-to/multi.md
Comment (dfellis) [inline docs/how-to/multi.md:15]:
```suggestion
      "vmType": "t3.medium"
```

File: docs/how-to/azure.md
Comment (dfellis) [inline docs/how-to/azure.md:39]:
```suggestion
    "vmType": "Standard_B1ls"
```

File: docs/reference/project-structure.md
Comment (dfellis) [inline docs/reference/project-structure.md:61]: Should we note that at the moment we do not support AWS's ARM VMs?

File: docs/background/autoscaling.md
Comment (dfellis) [reply docs/background/autoscaling.md:3]: So the link I shared explicitly calls it "Explanation" not "Background Information." It's possible that they're still feeling out what that quadrant is useful for, themselves. I think "Explanation" is a better fit and more useful in general because it has a better focus on "what questions would the user be asking?" rather than "here's a dump of random crap that we think is relevant background."

Moving on, I don't entirely agree with the assertion that it should be in the homepage. You can have a paragraph or two on what Anycloud is best for, but a deep dive article on *why* it really works better in certain circumstances deserves more attention/clarity, and is a good thing to share out.

Also, these two files are still placeholders, so perhaps we should just delete the whole section until we're ready to put things here?

File: docs/how-to/azure.md
Comment (cdmistman) [inline docs/how-to/azure.md:3]: nit:
```suggestion
## Enable programmatic Azure access for AnyCloud
```

File: docs/reference/project-structure.md
Comment (cdmistman) [inline docs/reference/project-structure.md:61]:
```suggestion
Each cloud provider will have a different possible values for `region` and `vmType`.
```

Comment (depombo) [reply docs/reference/project-structure.md]: Yeah

File: docs/reference/project-structure.md
Comment (depombo) [reply docs/reference/project-structure.md:13]: The link that goes nowhere has been removed. Adding a link to the section seems like an overkill give the sections are so short

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:106]: fixed

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:102]: No, that PR has not been landed

File: docs/summary.md
Comment (depombo) [reply docs/summary.md:10]: > Would reorder this to have a sense of structure like options, deploys, multicloud:
> 
> * [Define a custom app ID](how-to/custom-app.md)
> * [Pass environment variables](how-to/env.md)
> * [Deploy to GCP](how-to/gcp.md)
> * [Deploy to Azure](how-to/azure.md)
> * [Deploy an app to multiple clouds and/or regions](how-to/multi.md)
> 
> Understand that AWS still in progress, right?

I would put environment variables on top of GCP, but not Azure.

Comment (depombo) [reply docs/summary.md:10]: > +1 on this, still missing the `Deploy to AWS` section. It's mostly copy-paste from the tutorial, but it should be explicitly called out separately for someone looking for it that doesn't want to go through "the whole tutorial"

`Deploy to AWS` is part of the tutorial and called out explicitly at the beginning of the docs so it seems unnecessary to copy paste or call it out explicitly again?

File: docs/background/autoscaling.md
Comment (depombo) [reply docs/background/autoscaling.md:3]: I'm game to remove the section until we're ready to put things here

File: docs/tutorial.md
Comment (depombo) [reply docs/tutorial.md:117]: btw @dfellis in order to display `status` in the column we need to call the status/health control port outside of the upgrading a cluster, can you address https://github.com/alantech/deploy/pull/62#discussion_r599958706 in a follow up PR?

**Review by cdmistman — APPROVED** [review]

File: docs/summary.md
Comment (dfellis) [reply docs/summary.md]: Blocking on this one.

**Review by dfellis — CHANGES_REQUESTED** [review]
https://github.com/alantech/anycloud/pull/34#discussion_r601801234

File: docs/summary.md
Comment (depombo) [reply docs/summary.md]: I already addressed this here https://github.com/alantech/anycloud/pull/34/files#r600153220. I couldn't even find this message thread again afterwards. To be clear this absolutely seems like and not something that should be blocking and I already expressed that it is wasteful and overwhelming to go through the amount of comments on this PR and I would address in follow up PR.

**Comment by depombo** [comment]
> [#34 (comment)](https://github.com/alantech/anycloud/pull/34#discussion_r601801234)

https://github.com/alantech/anycloud/pull/34#discussion_r601803316

File: docs/summary.md
Comment (dfellis) [reply docs/summary.md]: That's not what matters. End users will come into these docs not knowing the structure. They will have a question they want to get and answer for and they will do their own searching through the docs for it.

If their question is: "How do I use this with AWS?" and they go to the "How to deploy..." section and don't see it listed,they could justifiably come to the conclusion that we don't support AWS.

So this is a missing piece that I consider a blocking concern and I cannot fathom why you don't see it the same way.

Comment (depombo) [reply docs/summary.md]: I already addressed that before you made your blocking comment/review in this commit https://github.com/alantech/anycloud/pull/34/commits/0e1eebee1c8e1fec64ea16a16c0de5235ff6e8e8
as I mentioned before but I pasted the wrong link. As for what matters, imo is not something that I believe should be blocking as I mentioned in discord https://discord.com/channels/682284785519165454/793276115191332904/824732609103659038. I asked if I could address in follow up PRs since I can't really keep track of everything that is going on here and would address it separately in a follow up PR. This is something we have done often in the past so not sure why we are diverging here, but again I already addressed it. The fact that you missed my commit kinda proves the difficulty I am also having with Github and this PR in particular

Comment (dfellis) [reply docs/summary.md]: That solution wasn't missed. It's just below my bar? This documentation is *the way* people will be learning about `anycloud` and how to use it, so I'm sorry if you think I'm being picky, but I believe we need to make good first impressions here.

I was going to suggest including an animated gif of using the `anycloud` CLI on the main README, but I decided to be quiet about that for the first real version of the docs.

Comment (depombo) [reply docs/summary.md]: I am happy to address your comment and make the documentation better I just thought you hadn't seen that. Per my question, I want to address it this PR is getting nightmarish to track so I am asking if I address that in a follow up PR?

**Review by dfellis — APPROVED** [review]
# 😑 lets move on

File: docs/tutorial.md
Comment (dfellis) [reply docs/tutorial.md:117]: https://github.com/alantech/deploy/issues/64

**Pull request merged**
